drivers/input/input_cst816s.c

```diff
@@ -288,40 +240,33 @@ static int cst816s_init(const struct device *dev)
 		return ret;
 	}
 
-	ret = gpio_pin_interrupt_configure_dt(&config->int_gpio,
-			GPIO_INT_EDGE_TO_ACTIVE);
+	ret = gpio_pin_interrupt_configure_dt(&config->int_gpio, GPIO_INT_EDGE_TO_ACTIVE);
 	if (ret < 0) {
 		LOG_ERR("Could not configure interrupt GPIO interrupt.");
 		return ret;
 	}
 
-	gpio_init_callback(&data->int_gpio_cb, cst816s_isr_handler,
-			   BIT(config->int_gpio.pin));
+	gpio_init_callback(&data->int_gpio_cb, cst816s_isr_handler, BIT(config->int_gpio.pin));
+	gpio_add_callback(config->int_gpio.port, &data->int_gpio_cb);
```

Comment:
Actually could you add error checking on this one? Saves a future coverity violation down the road https://github.com/zephyrproject-rtos/zephyr/issues/60475

Reply:
Ah, good catch. The other PR will need this also :^)